feature-wallet-impl/src/main/java/jp/co/soramitsu/feature_wallet_impl/data/mappers/TransactionMappers.kt

```diff
@@ -1,102 +1,130 @@
 import jp.co.soramitsu.feature_wallet_api.domain.model.WalletAccount
 import jp.co.soramitsu.feature_wallet_api.domain.model.amountFromPlanks
-import jp.co.soramitsu.feature_wallet_api.domain.model.planksFromAmount
-import jp.co.soramitsu.feature_wallet_impl.data.network.model.response.TransactionRemote
-import jp.co.soramitsu.feature_wallet_impl.presentation.model.TransactionModel
+import jp.co.soramitsu.feature_wallet_impl.data.network.model.response.SubqueryHistoryElementResponse
 
-fun mapTransactionStatusToTransactionStatusLocal(status: Transaction.Status) = when (status) {
-    Transaction.Status.PENDING -> TransactionLocal.Status.PENDING
-    Transaction.Status.COMPLETED -> TransactionLocal.Status.COMPLETED
-    Transaction.Status.FAILED -> TransactionLocal.Status.FAILED
-}
 
-fun mapTransactionStatusLocalToTransactionStatus(status: TransactionLocal.Status) = when (status) {
-    TransactionLocal.Status.PENDING -> Transaction.Status.PENDING
-    TransactionLocal.Status.COMPLETED -> Transaction.Status.COMPLETED
-    TransactionLocal.Status.FAILED -> Transaction.Status.FAILED
+fun mapOperationStatusToOperationLocalStatus(status: Operation.Status) = when (status) {
+    Operation.Status.PENDING -> OperationLocal.Status.PENDING
+    Operation.Status.COMPLETED -> OperationLocal.Status.COMPLETED
+    Operation.Status.FAILED -> OperationLocal.Status.FAILED
 }
 
-fun mapTransactionToTransactionModel(transaction: Transaction): TransactionModel {
-    return with(transaction) {
-        TransactionModel(
+fun mapOperationToOperationLocalDb(operation: Operation, source: OperationLocal.Source): OperationLocal {
+    with(operation) {
+        val amount = this.operation.getOperationAmount()
+        val fee = this.operation.getOperationFee()
+
+        return OperationLocal(
             hash = hash,
-            type = tokenType,
-            senderAddress = senderAddress,
-            recipientAddress = recipientAddress,
-            accountName = accountName,
-            isIncome = isIncome,
-            date = date,
-            amount = amount,
-            status = status,
-            fee = fee,
-            total = total
+            address = address,
+            time = time * 1000,
+            tokenType = mapTokenTypeToTokenTypeLocal(tokenType),
+            type = this.operation.header,
+            call = this.operation.subheader,
+            amount = amount?.toBigInteger(),
+            sender = (this.operation as? Operation.Operation.Transfer)?.sender,
+            receiver = (this.operation as? Operation.Operation.Transfer)?.receiver,
+            fee = fee?.toBigInteger(),
+            isReward = (this.operation as? Operation.Operation.Reward)?.isReward,
+            era = (this.operation as? Operation.Operation.Reward)?.era,
+            validator = (this.operation as? Operation.Operation.Reward)?.validator,
+            success = (this.operation as? Operation.Operation.Extrinsic)?.success,
```

Comment:
Probably it will look cleaner if we will do
```
when(this.operation) {
is Reward -> reward mapping
...
}
```
What do you think?

Reply:
Tried different approach. How about it?

Reply:
It seems like you forgot to push commit with this changes, since the code is unchanged